### ISSUE TITLE
Add module decoupleR to nf-core/modules

### ISSUE THREAD

**Issue by atrigila** [nf-core/differentialabundance#350]
### Description of feature

Add module decoupleR (and progeny)to nf-core/modules

**Comment by atrigila** [comment]
Decoupler [is available in nf-core/modules](https://github.com/nf-core/modules/blob/b48e64d5982a587e57e9d3fba585d852c68ac512/modules/nf-core/decoupler/decoupler/main.nf) but needs modifications to accept progeny data

**Comment by grst** [comment]
Yes, please take a look at the decoupler module and try to make it as generic as possible. 
Integrating it into the pipeline is currently blocked by the development of a subworkflow for enrichment analyses (#384 )
